Formula/m/micromamba.rb

```diff
@@ -49,7 +50,6 @@ def install
       -DBUILD_LIBMAMBA=ON
       -DBUILD_SHARED=ON
       -DBUILD_MICROMAMBA=ON
-      -DMICROMAMBA_LINKAGE=DYNAMIC
```

Comment:
Let's just ship `mamba` instead if that's the binary that supports shared linkage.

Reply:
Shouldn't the formula be named mamba then?

Reply:
Sure, we could do that. Though maybe we should just add a new `mamba` formula and deprecate this one.

I'd have suggested a formula rename, but you've said that `micromamba` and `mamba` do not expose the same interface.

Reply:
I'm only interested in the new interface, the old "conda-compatible" one doesn't really have any reason to exist, IMO, now that conda has a (default) libmamba solver. But the micromamba interface is cleaner and using it is better in CI. I'll see if I can get any devs to comment.

Reply:
Ah, I'm now wrong:

> Mamba 2.0 Changes
> Command Line Executables
> 
> Mamba (executable)
> 
> mamba, previously a Python executable mixing libmambapy, conda, and code to bridge both project is being replace by a fully C++ executable based on libmamba solely.
> 
> It now presents the same user interface and experience as micromamba.

https://mamba.readthedocs.io/en/latest/developer_zone/changes-2.0.html

Okay, that's a breaking change, and so I think renaming to mamba and just providing mamba is likely the correct path forward.

In my defense, the docs still claim it has a different interface in several places.

Reply:
Can I just rename the formula in the PR? Or is there more to do?

Reply:
If it's just the same now can't we just build `mamba` and create a symlink?

Reply:
Yes, but should the formula still be renamed? Or an alias added (since both are shipped in one formula, that seems best?)?

Reply:
Alias is probably simpler.